tools/roslaunch/src/roslaunch/depends.py

```diff
@@ -146,32 +146,39 @@ def _parse_launch(tags, launch_file, file_deps, verbose, context):
                 sub_launch_file = resolve_args(tag.attributes['file'].value, context)
             except KeyError as e:
                 raise RoslaunchDepsException("Cannot load roslaunch <%s> tag: missing required attribute %s.\nXML is %s"%(tag.tagName, str(e), tag.toxml()))
-                
-            if verbose:
-                print("processing included launch %s"%sub_launch_file)
-
-            # determine package dependency for included file
-            sub_pkg = rospkg.get_package_name(os.path.dirname(os.path.abspath(sub_launch_file)))
-            if sub_pkg is None:
-                print("ERROR: cannot determine package for [%s]"%sub_launch_file, file=sys.stderr)
-            
-            if sub_launch_file not in file_deps[launch_file].includes:
-                file_deps[launch_file].includes.append(sub_launch_file)
-            if launch_file_pkg != sub_pkg:            
-                file_deps[launch_file].pkgs.append(sub_pkg)
-            
-            # recurse
-            file_deps[sub_launch_file] = RoslaunchDeps()
-            try:
-                dom = parse(sub_launch_file).getElementsByTagName('launch')
-                if not len(dom):
-                    print("ERROR: %s is not a valid roslaunch file"%sub_launch_file, file=sys.stderr)
-                else:
-                    launch_tag = dom[0]
-                    sub_context = _parse_subcontext(tag.childNodes, context)
-                    _parse_launch(launch_tag.childNodes, sub_launch_file, file_deps, verbose, sub_context)
-            except IOError as e:
-                raise RoslaunchDepsException("Cannot load roslaunch include '%s' in '%s'"%(sub_launch_file, launch_file))
+
+            # Check if an empty file is included, and skip if so.  This will allow a default-empty <include> inside a
+            # conditional to pass
+            if sub_launch_file == '':
+                print("WARNING: empty <include> in %s. Skipping <include> of %s" % (launch_file,
```

Comment:
Works for me.  Because an empty launch include is *potentially* problematic I think a strong notification would be preferable, but really I'm good either way.

Note however that by running `rosrun roslaunch roslaunch-check /path/to/files`, this would never be triggered, since [roslaunch-check does not take a verbosity arg](https://github.com/ros/ros_comm/blob/kinetic-devel/tools/roslaunch/scripts/roslaunch-check#L62), and `verbose` defaults to false.  Since this command is a part of our pre-commit workflow, I may address this in a later merge request.